src/Drupal/Commands/core/BatchCommands.php

```diff
@@ -13,6 +13,7 @@ class BatchCommands extends DrushCommands
      * @aliases batch-process
      * @param $batch_id The batch id that will be processed.
      * @hidden
+     * @kernel update
```

Comment:
This is obviously not correct, but it is an attempt at getting the batch process to use the `UpdateKernel`. This is not working unfortunately, for a reason I don't understand yet the `batch-process` command is not found by the Symfony Console `Application`. It throws a `CommandNotFoundException` with message `Command "batch-process" is not defined.`. This is caught in `\Drush\Application::bootstrapAndFind()` which will then proceed to raise the bootstrap level to maximum, booting the wrong kernel.

Reply:
I think the best way to be successful here would be to factor out the batch process execution code into a separate class, and use that class from the batch process command and the updatedb command.